In [1]:
import pandas as pd

In [2]:
data1 = pd.read_excel('광운대 2022 인기도서 100개.xlsx')
data2 = pd.read_excel('광운대 2022 인기도서 300개.xlsx')
data3 = pd.read_excel('광운대학교 지정 도서.xlsx')
data4 = pd.read_excel('사서 추천 목록.xlsx')

# No. 열 제거, 필요한 열만 추출
- NO.,서명, 저자, 발행처, 출판연도, 이용건수 열
- 분석에 필요한 열만 추출하면 좋을 듯

In [3]:
data1 = data1[['서명','저자','발행처','이용건수']]
data1

서명                  저자   
0                                  불편한 편의점 : 김호연 장편소설                 김호연  \
1                                      파친코 : 이민진 장편소설                 이민진   
2                       (윤성우의 열혈) C 프로그래밍 / 개정판[실은2판]                 윤성우   
3      물고기는 존재하지 않는다 : 상실, 사랑 그리고 숨어 있는 삶의 질서에 관한 이야기        Miller, Lulu   
4                                               대학물리학   Serway, Raymond A   
..                                                ...                 ...   
95  파이썬 머신러닝 완벽 가이드 : 다양한 캐글 예제와 함께 기초 알고리즘부터 최신 기...                 권철민   
96      팩트풀니스 : 우리가 세상을 오해하는 10가지 이유와 세상이 생각보다 괜찮은 이유       Rosling, Hans   
97                               옷소매 붉은 끝동 : 강미강 장편소설                 강미강   
98            사피엔스 : 유인원에서 사이보그까지, 인간 역사의 대담하고 위대한 질문  Harari, Yuval Noah   
99                                    저주 토끼 : 정보라 소설집                 정보라   

           발행처  이용건수  
0        나무옆의자    90  
1         문학사상    77  
2       오렌지미디어    55  
3            곰    54  
4          북스힐    51  
..         ...   ...  
95        위키북스    18  
96         김영사    17  
97         청어람    17  
98         김영사    17  
99  아작 : 디자인콤마    17  

[100 rows x 4 columns]

In [24]:
data2 = data2[['서명','저자','발행처','이용건수']]
data2 = data2.rename(columns={'발행처':'출판사'})
data2

KeyError: "['발행처'] not in index"

In [25]:
data2.to_csv('광운대 인기도서.csv', encoding='utf-8-sig', index=False)

In [5]:
data3 = data3[['교과목명','대학']]
data3

교과목명          대학
0      건축계획및법규1 - 01        공과대학 
1         건축동역학 - 01        공과대학 
2    건축설계및도서작성1 - 02        공과대학 
3    건축설계및도서작성1 - 03        공과대학 
4         건축환경1 - 01        공과대학 
..                ...         ...
451  소프트웨어프로젝트1 - 01   소프트웨어융합대학 
452    시스템프로그래밍 - 01   소프트웨어융합대학 
453      신호및시스템 - 01   소프트웨어융합대학 
454       컴퓨터비젼 - 01   소프트웨어융합대학 
455       확률및통계 - 01   소프트웨어융합대학 

[456 rows x 2 columns]

In [18]:
data4 = data4[['상품명','인물','출판사','분야']]
data4

상품명            인물              출판사       분야
0                 봄은 언제나 찾아온다      데이비드 호크니             시공아트  예술/대중문화
1                       가족 공부           최광현        EBS BOOKS       인문
2              나로서 충분히 괜찮은 사람           김재식             북로망스    시/에세이
3            나는 불안할 때 논어를 읽는다            판덩             미디어숲     자기계발
4     언제 올지 모를 희망 말고 지금 행복했으면           송정림            자음과모음    시/에세이
5                      고립의 시대        노리나 허츠          웅진지식하우스    정치/사회
6                  지구를 살리는 옷장           박진영               창비    정치/사회
7                     안녕을 위하여           이승연           초록비책공방       인문
8             마음은 괜찮냐고 시가 물었다           황인환  웨일북(whalebooks)       인문
9          그럴수록 우리에겐 친구가 필요하다     이름트라우트 타르              갤리온     자기계발
10              나에게 오늘을 선물합니다           김나위               다연    시/에세이
11               여덟 건의 완벽한 살인        피터 스완슨              푸른숲       소설
12                      8초 인류        리사 이오띠             미래의창       과학
13              화성 탐사선을 탄 걸리버           곽재식             문학수첩       과학
14                     코끝의 언어       주드 스튜어트               윌북       인문
15                    Z의 스마트폰           박준영            쌤앤파커스    경제/경영
16           소년은 어떻게 과학자가 되었나       오구리 히로시            바다출판사       과학
17                       은하환담           곽재식               달다       소설
18              도시로 보는 동남아시아사           강희정               사우    역사/문화
19                   유령의 마음으로           임선우              민음사       소설
20                     나는, 휴먼        주디스 휴먼              사계절    정치/사회
21  경이로운 수 이야기: 영, 무한, 공포의 13  알브레히트 보이텔슈파허             해리북스       과학
22                   마음챙김 미술관           김소울            타인의사유       인문
23                      숨을 참다           김종진            후마니타스    정치/사회
24                 가구, 집을 갖추다           김지수               싱긋       인문
25                    낯선 삼일운동           정병욱            역사비평사    역사/문화
26                    잠자는 추억들      파트릭 모디아노             문학동네       소설
27                      요즘 애들      앤 헬렌 피터슨          알에이치코리아    정치/사회
28          내가 원하는 것을 나도 모를 때           전승환             다산초당       인문
29              바람이 강하게 불고 있다        미우라 시온              청미래       소설
30                 청년 도배사 이야기           배윤슬               궁리    시/에세이
31                        풍덩!           우지현           위즈덤하우스    시/에세이
32    밀레니얼은 어떻게 배우고 일하며 성장하는가           이상준             다른상상     자기계발
33         나를 힘들게 한 건 언제나 나였다        데일 카네기               예문     자기계발
34               꾸역꾸역이 뭐 어때서!           윤진오             바이북스     자기계발
35           불안해 보여서 불안한 당신에게           한창욱             레몬북스     자기계발
36                      제트코노미        제이슨 도시            서울문화사    경제/경영
37           결국 Z세대가 세상을 지배한다           김용섭             퍼블리온    경제/경영
38            삶은 속도가 아니라 방향이다            수영            루이앤휴잇     자기계발
39           보이지 않는 것에 의미가 있다           김혜남              포르체       인문
40                 헌책방 기담 수집가           윤성근            프시케의숲    시/에세이
41                 화려한 화학의 시대   프랭크 A. 폰 히펠               까치       과학
42               현대 한국어로 철학하기           신우승              메멘토       인문
43                      잔류 인구       엘리자베스 문              푸른숲       소설
44                     취향의 경제           유승호               따비    경제/경영
45    식물 심고 그림책 읽으며 아이들과 열두 달           이태용             세로북스    시/에세이
46                 과일로 읽는 세계사           윤덕노            타인의사유    역사/문화
47        예민한 사람을 위한 좋은 심리 습관          캐린 홀             빌리버튼     자기계발
48               조선의 위기 대응 노트           김준태              민음사    역사/문화
49              법정에서 못다 한 이야기           박형남            휴머니스트    정치/사회
50                      다크데이터       데이비드 핸드             더퀘스트       과학
51            보라선 열차와 사라진 아이들       디파 아나파라              북로드       소설
52                 플라멩코 추는 남자           허태연             다산책방       소설
53                   리얼 월드 러닝           김하늬             푸른들녘      청소년
54                카이사르의 마지막 숨           샘 킨              해나무       과학
55                 나는 미술관에 간다           김영애           마로니에북스  예술/대중문화

In [20]:
data4 = data4.rename(columns={'상품명' : '서명', '인물':'저자'})
data4

서명            저자              출판사       분야
0                 봄은 언제나 찾아온다      데이비드 호크니             시공아트  예술/대중문화
1                       가족 공부           최광현        EBS BOOKS       인문
2              나로서 충분히 괜찮은 사람           김재식             북로망스    시/에세이
3            나는 불안할 때 논어를 읽는다            판덩             미디어숲     자기계발
4     언제 올지 모를 희망 말고 지금 행복했으면           송정림            자음과모음    시/에세이
5                      고립의 시대        노리나 허츠          웅진지식하우스    정치/사회
6                  지구를 살리는 옷장           박진영               창비    정치/사회
7                     안녕을 위하여           이승연           초록비책공방       인문
8             마음은 괜찮냐고 시가 물었다           황인환  웨일북(whalebooks)       인문
9          그럴수록 우리에겐 친구가 필요하다     이름트라우트 타르              갤리온     자기계발
10              나에게 오늘을 선물합니다           김나위               다연    시/에세이
11               여덟 건의 완벽한 살인        피터 스완슨              푸른숲       소설
12                      8초 인류        리사 이오띠             미래의창       과학
13              화성 탐사선을 탄 걸리버           곽재식             문학수첩       과학
14                     코끝의 언어       주드 스튜어트               윌북       인문
15                    Z의 스마트폰           박준영            쌤앤파커스    경제/경영
16           소년은 어떻게 과학자가 되었나       오구리 히로시            바다출판사       과학
17                       은하환담           곽재식               달다       소설
18              도시로 보는 동남아시아사           강희정               사우    역사/문화
19                   유령의 마음으로           임선우              민음사       소설
20                     나는, 휴먼        주디스 휴먼              사계절    정치/사회
21  경이로운 수 이야기: 영, 무한, 공포의 13  알브레히트 보이텔슈파허             해리북스       과학
22                   마음챙김 미술관           김소울            타인의사유       인문
23                      숨을 참다           김종진            후마니타스    정치/사회
24                 가구, 집을 갖추다           김지수               싱긋       인문
25                    낯선 삼일운동           정병욱            역사비평사    역사/문화
26                    잠자는 추억들      파트릭 모디아노             문학동네       소설
27                      요즘 애들      앤 헬렌 피터슨          알에이치코리아    정치/사회
28          내가 원하는 것을 나도 모를 때           전승환             다산초당       인문
29              바람이 강하게 불고 있다        미우라 시온              청미래       소설
30                 청년 도배사 이야기           배윤슬               궁리    시/에세이
31                        풍덩!           우지현           위즈덤하우스    시/에세이
32    밀레니얼은 어떻게 배우고 일하며 성장하는가           이상준             다른상상     자기계발
33         나를 힘들게 한 건 언제나 나였다        데일 카네기               예문     자기계발
34               꾸역꾸역이 뭐 어때서!           윤진오             바이북스     자기계발
35           불안해 보여서 불안한 당신에게           한창욱             레몬북스     자기계발
36                      제트코노미        제이슨 도시            서울문화사    경제/경영
37           결국 Z세대가 세상을 지배한다           김용섭             퍼블리온    경제/경영
38            삶은 속도가 아니라 방향이다            수영            루이앤휴잇     자기계발
39           보이지 않는 것에 의미가 있다           김혜남              포르체       인문
40                 헌책방 기담 수집가           윤성근            프시케의숲    시/에세이
41                 화려한 화학의 시대   프랭크 A. 폰 히펠               까치       과학
42               현대 한국어로 철학하기           신우승              메멘토       인문
43                      잔류 인구       엘리자베스 문              푸른숲       소설
44                     취향의 경제           유승호               따비    경제/경영
45    식물 심고 그림책 읽으며 아이들과 열두 달           이태용             세로북스    시/에세이
46                 과일로 읽는 세계사           윤덕노            타인의사유    역사/문화
47        예민한 사람을 위한 좋은 심리 습관          캐린 홀             빌리버튼     자기계발
48               조선의 위기 대응 노트           김준태              민음사    역사/문화
49              법정에서 못다 한 이야기           박형남            휴머니스트    정치/사회
50                      다크데이터       데이비드 핸드             더퀘스트       과학
51            보라선 열차와 사라진 아이들       디파 아나파라              북로드       소설
52                 플라멩코 추는 남자           허태연             다산책방       소설
53                   리얼 월드 러닝           김하늬             푸른들녘      청소년
54                카이사르의 마지막 숨           샘 킨              해나무       과학
55                 나는 미술관에 간다           김영애           마로니에북스  예술/대중문화

In [21]:
data5.to_csv('사서 추천 정리.csv', encoding='utf-8-sig', index=False)

위 지정도서 데이터는 과목명만 잇고 도서명 없음
# 예지님이 지정도서는 1로 표시해주심

In [12]:
data5 = pd.read_excel('광운대 2022 인기도서 300개 지정도서 유무 추가.xlsx')
data5

No.                                                 서명   
0      1                                 불편한 편의점 : 김호연 장편소설  \
1      2                                     파친코 : 이민진 장편소설   
2      3                      (윤성우의 열혈) C 프로그래밍 / 개정판[실은2판]   
3      4     물고기는 존재하지 않는다 : 상실, 사랑 그리고 숨어 있는 삶의 질서에 관한 이야기   
4      5                                              대학물리학   
..   ...                                                ...   
295  296                                  (생활코딩!) 리액트 프로그래밍   
296  297               국제경제론 = International economics / 8판   
297  298     (Kreyszig) 공업수학. 상, 상미분방정식·선형대수·벡터미적분 / [실은2판]   
298  299                                         브람스를 좋아하세요   
299  300  (Do it!) 장고 + 부트스트랩 : 파이썬 웹 개발의 정석 = Do it! Dj...   

                    저자      발행처       출판연도  이용건수  지정도서  
0                  김호연    나무옆의자  2021-2022    90   NaN  
1                  이민진     문학사상       2018    77   NaN  
2                  윤성우   오렌지미디어       2010    55   NaN  
3         Miller, Lulu        곰       2021    54   NaN  
4    Serway, Raymond A      북스힐       2017    51   1.0  
..                 ...      ...        ...   ...   ...  
295                이고잉     위키북스       2021    11   NaN  
296                김인준       다산       2022    11   1.0  
297    Kreyszig, Erwin    텍스트북스       2020    11   NaN  
298   Sagan, Francoise      민음사       2008    11   NaN  
299                이성용  이지스퍼블리싱       2021    11   NaN  

[300 rows x 7 columns]

# 필요한 열만 추출

# 지정도서에 결측치 0으로 채움, 0이 지정도서 아닌 도서!

In [13]:
data5 = data5.fillna(0)
data5

No.                                                 서명   
0      1                                 불편한 편의점 : 김호연 장편소설  \
1      2                                     파친코 : 이민진 장편소설   
2      3                      (윤성우의 열혈) C 프로그래밍 / 개정판[실은2판]   
3      4     물고기는 존재하지 않는다 : 상실, 사랑 그리고 숨어 있는 삶의 질서에 관한 이야기   
4      5                                              대학물리학   
..   ...                                                ...   
295  296                                  (생활코딩!) 리액트 프로그래밍   
296  297               국제경제론 = International economics / 8판   
297  298     (Kreyszig) 공업수학. 상, 상미분방정식·선형대수·벡터미적분 / [실은2판]   
298  299                                         브람스를 좋아하세요   
299  300  (Do it!) 장고 + 부트스트랩 : 파이썬 웹 개발의 정석 = Do it! Dj...   

                    저자      발행처       출판연도  이용건수  지정도서  
0                  김호연    나무옆의자  2021-2022    90   0.0  
1                  이민진     문학사상       2018    77   0.0  
2                  윤성우   오렌지미디어       2010    55   0.0  
3         Miller, Lulu        곰       2021    54   0.0  
4    Serway, Raymond A      북스힐       2017    51   1.0  
..                 ...      ...        ...   ...   ...  
295                이고잉     위키북스       2021    11   0.0  
296                김인준       다산       2022    11   1.0  
297    Kreyszig, Erwin    텍스트북스       2020    11   0.0  
298   Sagan, Francoise      민음사       2008    11   0.0  
299                이성용  이지스퍼블리싱       2021    11   0.0  

[300 rows x 7 columns]

# 지정도서가 아닌 도서들만 추출

In [14]:
data5=data5[data5['지정도서']==0]
data5

No.                                                 서명   
0      1                                 불편한 편의점 : 김호연 장편소설  \
1      2                                     파친코 : 이민진 장편소설   
2      3                      (윤성우의 열혈) C 프로그래밍 / 개정판[실은2판]   
3      4     물고기는 존재하지 않는다 : 상실, 사랑 그리고 숨어 있는 삶의 질서에 관한 이야기   
5      6                 공정하다는 착각 : 능력주의는 모두에게 같은 기회를 제공하는가   
..   ...                                                ...   
294  295  (새)먼나라 이웃나라 : 이원복교수님과 함께 떠나는 세계 역사 기행!. 1-20 /...   
295  296                                  (생활코딩!) 리액트 프로그래밍   
297  298     (Kreyszig) 공업수학. 상, 상미분방정식·선형대수·벡터미적분 / [실은2판]   
298  299                                         브람스를 좋아하세요   
299  300  (Do it!) 장고 + 부트스트랩 : 파이썬 웹 개발의 정석 = Do it! Dj...   

                    저자          발행처       출판연도  이용건수  지정도서  
0                  김호연        나무옆의자  2021-2022    90   0.0  
1                  이민진         문학사상       2018    77   0.0  
2                  윤성우       오렌지미디어       2010    55   0.0  
3         Miller, Lulu            곰       2021    54   0.0  
5    Sandel, Michael J  와이즈베리 : 미래엔       2020    50   0.0  
..                 ...          ...        ...   ...   ...  
294                이원복          김영사  1999-2018    11   0.0  
295                이고잉         위키북스       2021    11   0.0  
297    Kreyszig, Erwin        텍스트북스       2020    11   0.0  
298   Sagan, Francoise          민음사       2008    11   0.0  
299                이성용      이지스퍼블리싱       2021    11   0.0  

[237 rows x 7 columns]

In [10]:
data5.shape

(237, 7)

In [27]:
data5 = data5[['서명','저자','발행처','이용건수']]
data5 = data5.rename(columns={'발행처':'출판사'})

In [28]:
data5.to_csv('지정도서 제외 광운대 인기도서.csv', encoding='utf-8-sig', index=False)

지정도서 아닌 광운대 2022 인기도서는 237개! 300개 중에 63개는 지정도서였다

# 광운대 지정도서 아닌 인기도서와 사서 추천 도서 살펴보기

In [11]:
data4

상품명            인물              출판사  발행(출시)일자
0                 봄은 언제나 찾아온다      데이비드 호크니             시공아트  20220124
1                       가족 공부           최광현        EBS BOOKS  20220518
2              나로서 충분히 괜찮은 사람           김재식             북로망스  20220510
3            나는 불안할 때 논어를 읽는다            판덩             미디어숲  20220330
4     언제 올지 모를 희망 말고 지금 행복했으면           송정림            자음과모음  20220415
5                      고립의 시대        노리나 허츠          웅진지식하우스  20211119
6                  지구를 살리는 옷장           박진영               창비  20220425
7                     안녕을 위하여           이승연           초록비책공방  20220910
8             마음은 괜찮냐고 시가 물었다           황인환  웨일북(whalebooks)  20211207
9          그럴수록 우리에겐 친구가 필요하다     이름트라우트 타르              갤리온  20220314
10              나에게 오늘을 선물합니다           김나위               다연  20220615
11               여덟 건의 완벽한 살인        피터 스완슨              푸른숲  20220411
12                      8초 인류        리사 이오띠             미래의창  20220418
13              화성 탐사선을 탄 걸리버           곽재식             문학수첩  20220707
14                     코끝의 언어       주드 스튜어트               윌북  20220425
15                    Z의 스마트폰           박준영            쌤앤파커스  20220701
16           소년은 어떻게 과학자가 되었나       오구리 히로시            바다출판사  20220630
17                       은하환담           곽재식               달다  20220325
18              도시로 보는 동남아시아사           강희정               사우  20220405
19                   유령의 마음으로           임선우              민음사  20220325
20                     나는, 휴먼        주디스 휴먼              사계절  20220318
21  경이로운 수 이야기: 영, 무한, 공포의 13  알브레히트 보이텔슈파허             해리북스  20220318
22                   마음챙김 미술관           김소울            타인의사유  20220312
23                      숨을 참다           김종진            후마니타스  20220314
24                 가구, 집을 갖추다           김지수               싱긋  20220214
25                    낯선 삼일운동           정병욱            역사비평사  20220228
26                    잠자는 추억들      파트릭 모디아노             문학동네  20220325
27                      요즘 애들      앤 헬렌 피터슨          알에이치코리아  20211025
28          내가 원하는 것을 나도 모를 때           전승환             다산초당  20200108
29              바람이 강하게 불고 있다        미우라 시온              청미래  20220118
30                 청년 도배사 이야기           배윤슬               궁리  20210705
31                        풍덩!           우지현           위즈덤하우스  20210618
32    밀레니얼은 어떻게 배우고 일하며 성장하는가           이상준             다른상상  20201020
33         나를 힘들게 한 건 언제나 나였다        데일 카네기               예문  20211217
34               꾸역꾸역이 뭐 어때서!           윤진오             바이북스  20210210
35           불안해 보여서 불안한 당신에게           한창욱             레몬북스  20210825
36                      제트코노미        제이슨 도시            서울문화사  20210111
37           결국 Z세대가 세상을 지배한다           김용섭             퍼블리온  20210810
38            삶은 속도가 아니라 방향이다            수영            루이앤휴잇  20220103
39           보이지 않는 것에 의미가 있다           김혜남              포르체  20211208
40                 헌책방 기담 수집가           윤성근            프시케의숲  20211206
41                 화려한 화학의 시대   프랭크 A. 폰 히펠               까치  20211117
42               현대 한국어로 철학하기           신우승              메멘토  20220207
43                      잔류 인구       엘리자베스 문              푸른숲  20211029
44                     취향의 경제           유승호               따비  20211130
45    식물 심고 그림책 읽으며 아이들과 열두 달           이태용             세로북스  20211102
46                 과일로 읽는 세계사           윤덕노            타인의사유  20211115
47        예민한 사람을 위한 좋은 심리 습관          캐린 홀             빌리버튼  20211029
48               조선의 위기 대응 노트           김준태              민음사  20210813
49              법정에서 못다 한 이야기           박형남            휴머니스트  20211108
50                      다크데이터       데이비드 핸드             더퀘스트  20211005
51            보라선 열차와 사라진 아이들       디파 아나파라              북로드  20211119
52                 플라멩코 추는 남자           허태연             다산책방  20210916
53                   리얼 월드 러닝           김하늬             푸른들녘  20210819
54                카이사르의 마지막 숨           샘 킨              해나무  20210830
55                 나는 미술관에 간다           김영애       

In [12]:
data5 = data5[['서명','저자','발행처','출판연도']]
data5

서명                 저자   
0                                   불편한 편의점 : 김호연 장편소설                김호연  \
1                                       파친코 : 이민진 장편소설                이민진   
2                        (윤성우의 열혈) C 프로그래밍 / 개정판[실은2판]                윤성우   
3       물고기는 존재하지 않는다 : 상실, 사랑 그리고 숨어 있는 삶의 질서에 관한 이야기       Miller, Lulu   
5                   공정하다는 착각 : 능력주의는 모두에게 같은 기회를 제공하는가  Sandel, Michael J   
..                                                 ...                ...   
294  (새)먼나라 이웃나라 : 이원복교수님과 함께 떠나는 세계 역사 기행!. 1-20 /...                이원복   
295                                  (생활코딩!) 리액트 프로그래밍                이고잉   
297     (Kreyszig) 공업수학. 상, 상미분방정식·선형대수·벡터미적분 / [실은2판]    Kreyszig, Erwin   
298                                         브람스를 좋아하세요   Sagan, Francoise   
299  (Do it!) 장고 + 부트스트랩 : 파이썬 웹 개발의 정석 = Do it! Dj...                이성용   

             발행처       출판연도  
0          나무옆의자  2021-2022  
1           문학사상       2018  
2         오렌지미디어       2010  
3              곰       2021  
5    와이즈베리 : 미래엔       2020  
..           ...        ...  
294          김영사  1999-2018  
295         위키북스       2021  
297        텍스트북스       2020  
298          민음사       2008  
299      이지스퍼블리싱       2021  

[237 rows x 4 columns]

사서추천 파일 열명 바꿔보자

In [15]:
# 여러 개의 열 이름 변경
data4 = data4.rename(columns={'상품명': '서명', '인물': '저자', '발행(출시)일자':'발행처', '출판연도':'발행(출시)일자'})
data4

서명            저자              출판사  발행(출시)일자
0                 봄은 언제나 찾아온다      데이비드 호크니             시공아트  20220124
1                       가족 공부           최광현        EBS BOOKS  20220518
2              나로서 충분히 괜찮은 사람           김재식             북로망스  20220510
3            나는 불안할 때 논어를 읽는다            판덩             미디어숲  20220330
4     언제 올지 모를 희망 말고 지금 행복했으면           송정림            자음과모음  20220415
5                      고립의 시대        노리나 허츠          웅진지식하우스  20211119
6                  지구를 살리는 옷장           박진영               창비  20220425
7                     안녕을 위하여           이승연           초록비책공방  20220910
8             마음은 괜찮냐고 시가 물었다           황인환  웨일북(whalebooks)  20211207
9          그럴수록 우리에겐 친구가 필요하다     이름트라우트 타르              갤리온  20220314
10              나에게 오늘을 선물합니다           김나위               다연  20220615
11               여덟 건의 완벽한 살인        피터 스완슨              푸른숲  20220411
12                      8초 인류        리사 이오띠             미래의창  20220418
13              화성 탐사선을 탄 걸리버           곽재식             문학수첩  20220707
14                     코끝의 언어       주드 스튜어트               윌북  20220425
15                    Z의 스마트폰           박준영            쌤앤파커스  20220701
16           소년은 어떻게 과학자가 되었나       오구리 히로시            바다출판사  20220630
17                       은하환담           곽재식               달다  20220325
18              도시로 보는 동남아시아사           강희정               사우  20220405
19                   유령의 마음으로           임선우              민음사  20220325
20                     나는, 휴먼        주디스 휴먼              사계절  20220318
21  경이로운 수 이야기: 영, 무한, 공포의 13  알브레히트 보이텔슈파허             해리북스  20220318
22                   마음챙김 미술관           김소울            타인의사유  20220312
23                      숨을 참다           김종진            후마니타스  20220314
24                 가구, 집을 갖추다           김지수               싱긋  20220214
25                    낯선 삼일운동           정병욱            역사비평사  20220228
26                    잠자는 추억들      파트릭 모디아노             문학동네  20220325
27                      요즘 애들      앤 헬렌 피터슨          알에이치코리아  20211025
28          내가 원하는 것을 나도 모를 때           전승환             다산초당  20200108
29              바람이 강하게 불고 있다        미우라 시온              청미래  20220118
30                 청년 도배사 이야기           배윤슬               궁리  20210705
31                        풍덩!           우지현           위즈덤하우스  20210618
32    밀레니얼은 어떻게 배우고 일하며 성장하는가           이상준             다른상상  20201020
33         나를 힘들게 한 건 언제나 나였다        데일 카네기               예문  20211217
34               꾸역꾸역이 뭐 어때서!           윤진오             바이북스  20210210
35           불안해 보여서 불안한 당신에게           한창욱             레몬북스  20210825
36                      제트코노미        제이슨 도시            서울문화사  20210111
37           결국 Z세대가 세상을 지배한다           김용섭             퍼블리온  20210810
38            삶은 속도가 아니라 방향이다            수영            루이앤휴잇  20220103
39           보이지 않는 것에 의미가 있다           김혜남              포르체  20211208
40                 헌책방 기담 수집가           윤성근            프시케의숲  20211206
41                 화려한 화학의 시대   프랭크 A. 폰 히펠               까치  20211117
42               현대 한국어로 철학하기           신우승              메멘토  20220207
43                      잔류 인구       엘리자베스 문              푸른숲  20211029
44                     취향의 경제           유승호               따비  20211130
45    식물 심고 그림책 읽으며 아이들과 열두 달           이태용             세로북스  20211102
46                 과일로 읽는 세계사           윤덕노            타인의사유  20211115
47        예민한 사람을 위한 좋은 심리 습관          캐린 홀             빌리버튼  20211029
48               조선의 위기 대응 노트           김준태              민음사  20210813
49              법정에서 못다 한 이야기           박형남            휴머니스트  20211108
50                      다크데이터       데이비드 핸드             더퀘스트  20211005
51            보라선 열차와 사라진 아이들       디파 아나파라              북로드  20211119
52                 플라멩코 추는 남자           허태연             다산책방  20210916
53                   리얼 월드 러닝           김하늬             푸른들녘  20210819
54                카이사르의 마지막 숨           샘 킨              해나무  20210830
55                 나는 미술관에 간다           김영애        

# 두 데이터간 겹치는 내용 있나? 진행하려다가 기절

In [19]:
merged = pd.merge(data4, data5, on='서명')
merged

Empty DataFrame
Columns: [서명, 저자_x, 출판사, 발행(출시)일자, 저자_y, 발행처, 출판연도]
Index: []